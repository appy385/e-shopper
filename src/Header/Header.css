.header {
    display: flex;
    justify-content: space-between;
    background-color: black;
}
.header-tab {
    padding: 0.3rem;
    color: blanchedalmond;
}

.header-right {
    display: flex;
}