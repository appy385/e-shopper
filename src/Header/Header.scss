.header-dark {
    display: flex;
    justify-content: space-between;
    background-color: black;
    margin-top: 1rem;
    color: white;
}

.header-light {
    display: flex;
    justify-content: space-between;
    // background-color: #D3D3D3;
    background-color: white;
    // padding: 1rem;
    color: black;
}

.header-light > a {
    text-decoration: none;
}

.header-dark > a {
    text-decoration: none;
}

.header-tab-dark {
    padding: 0.3rem;
    color:white;
}
.header-tab-light{
    padding: 0.3rem;
    color: black;
}
.header-right {
    display: flex;
    flex-direction: row;
}

.header-basket-image {
    align-self: center;
    width: 3rem;
    // height: 1rem;

}

.header-basket-container {
    display: flex;
    flex-direction: row;
    background-color: grey;
    border-left: 0.1px solid grey;
}