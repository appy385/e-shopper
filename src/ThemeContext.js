import React from 'react';

export const Theme = {
  light: 'light',
  dark: 'dark',
};
export const ThemeContext = React.createContext(Theme.light);
