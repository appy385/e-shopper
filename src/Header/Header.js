import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ThemeContext } from '../ThemeContext';
import './Header.scss';

const Header = ({ value }) => {
  const theme = useContext(ThemeContext);
  const headerTab = (theme === 'dark') ? 'header-tab-dark' : 'header-tab-light';
  return (
    <div className={(theme === 'dark') ? 'header-dark' : 'header-light'}>
      <Link to="/"><p className={`${headerTab} eshopper`}>E-shopper</p></Link>
      <div className="header-right">
        <Link to="/order"><p className={`${headerTab} orders`}>All Orders</p></Link>
        <Link to="/cart">
          <p className={`${headerTab} items`}>{`Basket Items: ${value}`}</p>
        </Link>

      </div>
    </div>
  );
};

Header.propTypes = {
  value: PropTypes.number.isRequired,

};

export default Header;
