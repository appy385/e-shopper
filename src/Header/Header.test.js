import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Header from './Header';
import { Theme, ThemeContext } from '../ThemeContext';

describe(Header.name, () => {
  const mockValue = 0;
  test('should match snapshot for theme light', () => {
    const { container } = render(
      <ThemeContext.Provider value={Theme.light}>
        <Header value={mockValue} />
      </ThemeContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  test('should match snapshot for theme dark', () => {
    const { container } = render(
      <ThemeContext.Provider value={Theme.dark}>
        <Header value={mockValue} />
      </ThemeContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  test('should go to new route when E-shopper link is clicked', () => {
    render(<Header value={mockValue} />);
    const eshopperLink = screen.getByText('E-shopper');
    fireEvent.click(eshopperLink);
    expect(document.location.href).toBe('http://localhost/');
  });

  test('should go to new route when All Orders link is clicked ', () => {
    render(<Header value={mockValue} />);
    const allOrdersLink = screen.getByText('All Orders');
    fireEvent.click(allOrdersLink);
    expect(document.location.href).toBe('http://localhost/order');
  });

  test('should go to new route when Cart link is clicked ', () => {
    render(<Header value={mockValue} />);
    const cartLink = screen.getByText(`Basket Items: ${mockValue}`);
    fireEvent.click(cartLink);
    expect(document.location.href).toBe('http://localhost/cart');
  });
});
