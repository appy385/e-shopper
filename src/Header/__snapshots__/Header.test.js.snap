// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot for theme dark 1`] = `
<div>
  <div
    class="header-dark"
  >
    <a
      href="/"
    >
      <p
        class="header-tab-dark eshopper"
      >
        E-shopper
      </p>
    </a>
    <div
      class="header-right"
    >
      <a
        href="/order"
      >
        <p
          class="header-tab-dark orders"
        >
          All Orders
        </p>
      </a>
      <a
        href="/cart"
      >
        <p
          class="header-tab-dark items"
        >
          Basket Items: 0
        </p>
      </a>
    </div>
  </div>
</div>
`;

exports[`Header should match snapshot for theme light 1`] = `
<div>
  <div
    class="header-light"
  >
    <a
      href="/"
    >
      <p
        class="header-tab-light eshopper"
      >
        E-shopper
      </p>
    </a>
    <div
      class="header-right"
    >
      <a
        href="/order"
      >
        <p
          class="header-tab-light orders"
        >
          All Orders
        </p>
      </a>
      <a
        href="/cart"
      >
        <p
          class="header-tab-light items"
        >
          Basket Items: 0
        </p>
      </a>
    </div>
  </div>
</div>
`;
